solidity/src/libraries/MerkleTree.sol

```diff
@@ -40,26 +41,24 @@ library MerkleTree {
     /// `proofOffset` parameters. It is the caller's responsibility to ensure
     /// that the `encodedProof` is long enough to contain the proof and the
     /// `proofOffset` is not out of bound.
-    function isProofValid(bytes calldata encodedProof, uint256 proofOffset, bytes32 root, bytes calldata leafData)
+    function isProofValid(bytes calldata encodedProof, uint256 offset, bytes32 root, bytes calldata leafData)
         internal
         pure
         returns (bool valid, uint256 endOffset)
     {
         unchecked {
             bytes32 currentDigest = MerkleTree.leafHash(leafData);
-
-            uint8 proofSize = UnsafeCalldataBytesLib.toUint8(encodedProof, proofOffset);
-            proofOffset += 1;
-
-            for (uint256 i = 0; i < proofSize; i++) {
+            uint256 proofOffset = 0;
+            uint16 proofSizeArray = UnsafeCalldataBytesLib.toUint16(encodedProof, proofOffset);
+            proofOffset += 2;
```

Comment:
I made this change because in the previous version, we mixed up the idea of bytes length and array length. The proof is an array of bytes32 stored as bytes, while proofSize is the number of bytes in total in the proof. This mismatch cause the code to fail here. I add the parameter proofSizeArray, which indicates the length of the proof array to avoid this.